integrations/openrouter/src/haystack_integrations/components/generators/openrouter/chat/chat_generator.py

```diff
@@ -0,0 +1,193 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Callable, Dict, List, Optional
+
+from haystack import component, default_to_dict, logging
+from haystack.components.generators.chat import OpenAIChatGenerator
+from haystack.dataclasses import ChatMessage, StreamingChunk
+from haystack.tools.tool import Tool, _check_duplicate_tool_names
+from haystack.utils import serialize_callable
+from haystack.utils.auth import Secret
+
+logger = logging.getLogger(__name__)
+
+StreamingCallbackT = Callable[[StreamingChunk], None]
+
+
+@component
+class OpenRouterChatGenerator(OpenAIChatGenerator):
+    """
+    Enables text generation using OpenRouter generative models.
+    For supported models, see [OpenRouter docs](https://openrouter.ai/models).
+
+    Users can pass any text generation parameters valid for the OpenRouter chat completion API
+    directly to this component via the `generation_kwargs` parameter in `__init__` or the `generation_kwargs`
+    parameter in `run` method.
+
+    Key Features and Compatibility:
+    - **Primary Compatibility**: Designed to work seamlessly with the OpenRouter chat completion endpoint.
+    - **Streaming Support**: Supports streaming responses from the OpenRouter chat completion endpoint.
+    - **Customizability**: Supports all parameters supported by the OpenRouter chat completion endpoint.
+
+    This component uses the ChatMessage format for structuring both input and output,
+    ensuring coherent and contextually relevant responses in chat-based text generation scenarios.
+    Details on the ChatMessage format can be found in the
+    [Haystack docs](https://docs.haystack.deepset.ai/v2.0/docs/data-classes#chatmessage)
+
+    For more details on the parameters supported by the OpenRouter API, refer to the
+    [OpenRouter API Docs](https://openrouter.ai/docs/quickstart).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.generators.openrouter import OpenRouterChatGenerator
+    from haystack.dataclasses import ChatMessage
+
+    messages = [ChatMessage.from_user("What's Natural Language Processing?")]
+
+    client = OpenRouterChatGenerator()
+    response = client.run(messages)
+    print(response)
+
+    >>{'replies': [ChatMessage(content='Natural Language Processing (NLP) is a branch of artificial intelligence
+    >>that focuses on enabling computers to understand, interpret, and generate human language in a way that is
+    >>meaningful and useful.', role=<ChatRole.ASSISTANT: 'assistant'>, name=None,
+    >>meta={'model': 'openai/gpt-4o-mini', 'index': 0, 'finish_reason': 'stop',
+    >>'usage': {'prompt_tokens': 15, 'completion_tokens': 36, 'total_tokens': 51}})]}
+    ```
+    """
+
+    def __init__(
+        self,
+        api_key: Secret = Secret.from_env_var("OPENROUTER_API_KEY"),
```

Comment:
I think we've started to prefer keyword only arguments, so if @julian-risch agrees I think changing this to 
```suggestion
        self,
        *,
        api_key: Secret = Secret.from_env_var("OPENROUTER_API_KEY"),
```
would be a good idea